lib/ansible/executor/play_iterator.py

```diff
@@ -473,6 +473,13 @@ def mark_host_failed(self, host):
         self._host_states[host.name] = s
         self._play._removed_hosts.append(host.name)
 
+    def host_not_in_inventory(self, hostname):
+        display.debug("marking host %s failed, host not found in inventory: %s" % (hostname))
+        s = self._set_failed_state(self.FAILED_NONE)
```

Comment:
I don't think this would work. `_set_failed_state` expects a `HostState` object, `self.FAILED_NONE` is `int` so I think it would fail once `_set_failed_state` tries to access a `HostState` attribute.

Unsure but we probably need to create a `HostState` instance and set `state.fail_state = self.FAILED_NONE`?

Reply:
Actually, `FAILED_NONE` is default, so probably just `Hoststate(blocks=[])` or something like that?

Reply:
good catch